### PR TITLE
Toggle input

### LINKED ISSUE #387: Create large checkbox
As featured in the signup flow

### PR DESCRIPTION
**Pull request by alexpaxton**
Closes #387 

Rather than create a large checkbox specifically for the signup flow, I decided to create a more generic and customizable "toggle" component. Although the `Input` component currently supports a `checkbox` type I did not want to add more complexity to its API.

While I would suggest to anyone building a form to use the new `Toggle` component, I did not remove support for `InputType.Checkbox` within `Input` to prevent breaking changes

### Changes

- Introduce `Toggle` component that supports html "radio" and "checkbox" inputs
  - Can be `Outline` or `Solid` via `Appearance`
  - Has a dot by default, but can be customized to use any icon from `IconFont`
  - `size` and `color` can be customized (`Large` size will work for the signup checkbox)
- Abstract `PopoverType` into the more generic `Appearance`
  - "Appearance" seemed like a more intuitive name now that it is being used across 2 components, and potentially more
- Refactor `<SlideToggle.Label />` into a more discreet `<InputLabel />`
- Add `htmlFor` and `tabIndex` props to `InputLabel` so they can be configured for keyboard usage

### Screenshots

radio-input-tabbing
toggle-examples
checkbox-tab-mouse

### Checklist
Check all that apply

- [x] Updated documentation to reflect changes
- [x] Added entry to top of Changelog with link to PR (not issue)
- [x] Tests pass
- [x] Peer reviewed and approved
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: src/Components/Button/Documentation/Button.stories.tsx [hunk 1]
```diff
@@ -188,7 +188,7 @@ buttonComposedStories.add(
           ]
         }
         popoverType={
-          PopoverType[select('popoverType', mapEnumKeys(PopoverType), 'Solid')]
+          Appearance[select('popoverType', mapEnumKeys(Appearance), 'Solid')]
```
Comment (ebb-tide) [inline src/Components/Button/Documentation/Button.stories.tsx]: I'm finding `popoverType= Appearance[...]` a little confusing :(

File: src/Components/SlideToggle/Documentation/SlideToggle.stories.tsx [hunk 2]
```diff
@@ -21,7 +22,7 @@ import {
 
 // Notes
 import SlideToggleReadme from './SlideToggle.md'
-import SlideToggleLabelReadme from './SlideToggleLabel.md'
+// import SlideToggleLabelReadme from './SlideToggleLabel.md'
```
Comment (ebb-tide) [inline src/Components/SlideToggle/Documentation/SlideToggle.stories.tsx]: beep boop comment

Comment (alexpaxton) [reply src/Components/SlideToggle/Documentation/SlideToggle.stories.tsx]: Oh good catch!

File: src/Components/Button/Documentation/Button.stories.tsx (hunk 1 above)
Comment (alexpaxton) [reply src/Components/Button/Documentation/Button.stories.tsx]: I could rename the `popoverType` prop to `appearance` for ease of use

**Pull request merged**